File: 钙信号笔记/网格划分/二维网格划分程序/grid1/2d/npoch.f90
program aa
integer npoch(50000),sideface(3)

OPEN(1,FILE='SIDEGRIDT.dat')
READ(1,*)NBP
CLOSE(1)
OPEN(1,FILE='GRIDT.dat')
READ(1,*)NP
CLOSE(1)
do i = 1,nbp,1
   npoch(i)=1;
enddo
do i = nbp+1,np,1
npoch(i)=0;
enddo
open(1,file='SIDEFACE.dat')
read(1,*)nS
do i=1,nS
  read(1,*)(SIDEFACE(j),j=1,3)
  if (sideface(3) .ne. 1)then
    npoch(sideface(1))=sideface(3);
    npoch(sideface(2))=sideface(3);
  endif
enddo
close(1)

open(1,file='npoch.dat')
do i=1,np
write(1,*)npoch(i)
enddo
close(1)
end program

File: 纳米钙火花非结构网格（4通道）/myGrid程序/5_nopch/npoch.py
import numpy as npp
import math


def not_empty(s):
    return s and s.strip()


# 这个文件的作用是为每一个边界点定义一个属性，内部点是0，流进的边界点是1

def aa():
    npoch = npp.zeros(50000, dtype="int")
    sideface = npp.zeros(3, dtype="int")
    si_file = open("4RYRsidegridt.dat", "r")
    sicount = 0
    for line in si_file.readlines():
        sicount = sicount + 1
    nbp = sicount
    si_file.close()
    gri_file = open("4RYRgridt.dat", "r")
    gricount = 0
    for line in gri_file.readlines():
        gricount = gricount + 1
    np = gricount
    gri_file.close()
    for i in range(0, nbp):
        npoch[i] = 1

    for i in range(nbp, np):
        npoch[i] = 0

    sideface_file = open("4RYRsideface.dat", "r")
    sidefacecount = 0
    for line in sideface_file.readlines():
        current_line = list(filter(not_empty, line.strip("\n").split(" ")))
        sideface[0] = current_line[0]
        sideface[1] = current_line[1]
        sideface[2] = current_line[2]
        if sideface[2] != 1:
            npoch[sideface[0]-1] = sideface[2]
            npoch[sideface[1]-1] = sideface[2]
        sidefacecount=sidefacecount+1
    ns=sidefacecount
    sideface_file.close()
    npoch_file = open("4RYRnpoch.dat", "w")
    for i in range(0, np):
        npoch_file.write(str(npoch[i])+"\n")
    npoch_file.close()
aa()